contrib/macos/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.2)
```

Comment:
this would define another project just for macos. I think we should avoid this and work with components within the main project and use cpack (part of cmake) to install everything where its needs to be